courseManagementService/src/main/java/com/lms/courseManagementService/controller/ModuleController.java

```diff
@@ -0,0 +1,98 @@
+package com.lms.courseManagementService.controller;
+
+import com.lms.courseManagementService.dto.ModuleDTO;
+import com.lms.courseManagementService.exception.ModuleNotFoundException;
+import com.lms.courseManagementService.mapper.ModuleMapper;
+import com.lms.courseManagementService.model.Module;
+import com.lms.courseManagementService.response.ApiResponse;
+import com.lms.courseManagementService.service.ModuleService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/modules")
+public class ModuleController {
+    private final ModuleService moduleService;
+
+    @GetMapping("/all")
+    public ResponseEntity<List<ModuleDTO>> getAll() {
+        List<Module> modules = moduleService.getAllModules();
+        List<ModuleDTO> moduleDTOS = modules.stream()
+                .map(ModuleMapper :: toDTO)
+                .collect(Collectors.toList());
+
+        return ResponseEntity.ok(moduleDTOS);
+
+    }
+
+    @GetMapping("/{moduleId}")
+    public ResponseEntity<ApiResponse> getModuleById(@PathVariable("moduleId") Long moduleId) {
+        try {
+            Module module = moduleService.getModuleById(moduleId);
+            ModuleDTO moduleDTO = ModuleMapper.toDTO(module);
```

Comment:
Great, but if you called the mappers inside the service will be best practice to minimize the code inside the controller